elastic/kibana
# [License Management] NP migration

This PR migrates the License Management app to the new platform.

### How to test

- launch ES in trial mode `yarn es snapshot --license trial`
- keep a clean copy to be able to retest quickly `cp -aR .es/8.0.0 .es/8.0.0-clean`
- Launch Kibana
- Navigate to the App
- Click "Extend trial" and make sure we are directed to the correct URL to extend the trial
- Go back to the app and click "Revert to Basic license". A modal should open. Click "Confirm"
- The browser window should refresh, and once back on the app there should only be 2 options: "Update your license" and "Extend trial"
- The "Cross cluster replication" app should not appear in the menu on the left
- Click "Update your license" and drag & drop the platinum dev license 
- The browser window should refresh and you should see your active platinum license 

<img width="735" alt="Screenshot 2020-03-16 at 13 18 16" src="https://user-images.githubusercontent.com/2854616/76757812-a46c5e00-6788-11ea-9263-235b0165769e.png">

- Delete the license through the API. In a terminal window run `curl -XDELETE -v http://elastic:changeme@localhost:9200/_license`
- Manually refresh the Kibana browser window. You should see the Basic license back

<img width="743" alt="Screenshot 2020-03-16 at 13 20 23" src="https://user-images.githubusercontent.com/2854616/76757946-f44b2500-6788-11ea-85fa-140cf3e184a8.png">

- Stop Elasticsearch
- Delete the instance `rm -rf .es/8.0.0`
- Copy the clean instance `cp -aR .es/8.0.0-clean .es/8.0.0/`
- Restart ES `.es/8.0.0/bin/elasticsearch`
- Manually refresh the Kibana browser window. You should be back to the Trial license

<img width="1167" alt="Screenshot 2020-03-16 at 13 24 37" src="https://user-images.githubusercontent.com/2854616/76758239-98cd6700-6789-11ea-83b9-d5006599a1ec.png">
